tests/console/commands/test_install.py

```diff
@@ -4,32 +4,112 @@
 
 import pytest
 
+from poetry.core.masonry.utils.module import ModuleOrPackageNotFound
+
 
 if TYPE_CHECKING:
     from cleo.testers.command_tester import CommandTester
     from pytest_mock import MockerFixture
 
+    from poetry.poetry import Poetry
     from tests.types import CommandTesterFactory
+    from tests.types import ProjectFactory
+
+
+PYPROJECT_CONTENT = """\
+[tool.poetry]
+name = "simple-project"
+version = "1.2.3"
+description = "Some description."
+authors = [
+    "Python Poetry <tests@python-poetry.org>"
+]
+license = "MIT"
+readme = "README.rst"
+
+[tool.poetry.dependencies]
+python = "~2.7 || ^3.4"
+
+[tool.poetry.group.foo.dependencies]
+foo = "^1.0"
+
+[tool.poetry.group.bar.dependencies]
+bar = "^1.1"
+
+[tool.poetry.group.baz.dependencies]
+baz = "^1.2"
+
+[tool.poetry.group.bim.dependencies]
+bim = "^1.3"
+
+[tool.poetry.group.bam]
+optional = true
+
+[tool.poetry.group.bam.dependencies]
+bam = "^1.4"
+"""
 
 
 @pytest.fixture
-def tester(command_tester_factory: CommandTesterFactory) -> CommandTester:
+def poetry(project_factory: ProjectFactory) -> Poetry:
+    return project_factory(name="export", pyproject_content=PYPROJECT_CONTENT)
+
+
+@pytest.fixture
+def tester(
+    command_tester_factory: CommandTesterFactory, poetry: Poetry
+) -> CommandTester:
     return command_tester_factory("install")
 
 
+@pytest.mark.parametrize(
+    ("options", "groups", "with_root"),
+    [
+        case + (with_root,)
+        for case in [
+            ("", {"default", "foo", "bar", "baz", "bim"}),
+            ("--only default", {"default"}),
+            ("--only foo", {"foo"}),
+            ("--only foo,bar", {"foo", "bar"}),
+            ("--only bam", {"bam"}),
+            ("--with bam", {"default", "foo", "bar", "baz", "bim", "bam"}),
+            ("--without foo,bar", {"default", "baz", "bim"}),
+            ("--with foo,bar --without baz --without bim --only bam", {"bam"}),
```

Comment:
I'd like even more test cases:
`--default` (even if it is deprecated and will probably be removed before release)
`--with non-optional` (useless with)
`--with optional --without non-optional`
`--without default` (just in case even if it is no separate path in code)
`--without non-optional` (useless without)
`--with non-optional --without non-optional` (precedence, no effect)
`--with optional --without optional` (precedence, useless with)

Since you can combine `--with` and `--without`, it should be documented which one takes precedence. It's not yet docurmented anywhere, is it?